include/kvdk/engine.h

```diff
@@ -110,21 +111,39 @@ extern KVDKStatus KVDKSortedGet(KVDKEngine* engine, const char* collection,
                                 size_t collection_len, const char* key,
                                 size_t key_len, size_t* val_len, char** val);
 
-// For Hash Collection
-extern KVDKStatus KVDKHashSet(KVDKEngine* engine, const char* collection,
-                              size_t collection_len, const char* key,
-                              size_t key_len, const char* val, size_t val_len);
-extern KVDKStatus KVDKHashDelete(KVDKEngine* engine, const char* collection,
-                                 size_t collection_len, const char* key,
-                                 size_t key_len);
-extern KVDKStatus KVDKHashGet(KVDKEngine* engine, const char* collection,
-                              size_t collection_len, const char* key,
-                              size_t key_len, size_t* val_len, char** val);
+/// Hash //////////////////////////////////////////////////////////////////////
+extern KVDKStatus KVDKHashLength(KVDKEngine* engine, char const* key_data,
+                                 size_t key_len, size_t* len);
+extern KVDKStatus KVDKHashGet(KVDKEngine* engine, const char* key_data,
+                              size_t key_len, const char* field_data,
+                              size_t field_len, char** val_data,
+                              size_t* val_len);
+extern KVDKStatus KVDKHashSet(KVDKEngine* engine, const char* key_data,
+                              size_t key_len, const char* field_data,
+                              size_t field_len, const char* val_data,
+                              size_t val_len);
+extern KVDKStatus KVDKHashDelete(KVDKEngine* engine, const char* key_data,
+                                 size_t key_len, const char* field_data,
+                                 size_t field_len);
+/// HashIterator //////////////////////////////////////////////////////////////
+extern KVDKHashIterator* KVDKHashIteratorCreate(KVDKEngine* engine,
+                                                char const* key_data,
+                                                size_t key_len);
+extern void KVDKHashIteratorDestroy(KVDKHashIterator* iter);
+extern void KVDKHashIteratorPrev(KVDKHashIterator* iter);
+extern void KVDKHashIteratorNext(KVDKHashIterator* iter);
+extern void KVDKHashIteratorSeekToFirst(KVDKHashIterator* iter);
+extern void KVDKHashIteratorSeekToLast(KVDKHashIterator* iter);
```

Comment:
Maybe hash iterator do not need seek to first and seek to last. What is the last and the first meaning for in a hash collection?

Reply:
It's a random element with regard to key.
std::unordered_map has begin() and end().
Maybe we should SeekToFirst() internally when create the Iterator and only supply Next() and Valid()?